thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java

```diff
@@ -32,36 +39,261 @@
  *
  * {@link TiActivity} is responsible to manage the references
  */
-public enum PresenterSavior implements TiPresenterSavior {
+public class PresenterSavior implements TiPresenterSavior {
 
-    INSTANCE;
+    private static PresenterSavior INSTANCE;
 
     private static final String TAG = PresenterSavior.class.getSimpleName();
 
-    private final HashMap<String, TiPresenter> mPresenters = new HashMap<>();
+    static final String TI_ACTIVITY_PRESENTER_SCOPE_KEY = "ThirtyInch_presenter_scope_id";
+
+    @VisibleForTesting
+    Application.ActivityLifecycleCallbacks mLifecycleCallbacks;
+
+    @VisibleForTesting
+    final HashMap<String, ActivityScopedPresenters> mScopes = new HashMap<>();
+
+    private final HashMap<Activity, String> mScopeIdForActivity = new HashMap<>();
+
+    public static synchronized PresenterSavior getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new PresenterSavior();
+        }
+        return INSTANCE;
+    }
+
+    @VisibleForTesting
+    /*package*/ PresenterSavior() {
```

Comment:
I haven't ever seen smth like that before in this project.. So remove it please :)